### PR TITLE
Replace Hook with MethodHook where possible

### PR DESCRIPTION
**Pull request by deathbeam**
In order to cleanup the Hooks class a bit, replace runescape-client
Hook with MethodHook in mixins where possible/convenient.

Depends on !13
Depends on !14

Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

### CODE REVIEW

**Comment by Adam-** [comment]
This should probably have one commit per method moved. I suspect there will be things which no longer have to be exposed in rl-api as a result of this. Also the entire `@Hook` annotation can go away.

**Comment by deathbeam** [comment]
Alright. So that means for the performance-critical methods that cannot be just sent through eventBus i should just call methods in Hooks directly from mixins? In similar way as mouse/key listeners do atm?

**Comment by deathbeam** [comment]
Split each hook change to single commit, updated !13 with removal of `@Hook` annotation and used constructor injection for remaining hooks, so I call this update feature complete :d

File: runelite-client/src/main/java/net/runelite/client/callback/Hooks.java [hunk 1]
```diff
@@ -467,31 +464,6 @@ public static void menuOpened(Client client, int var1, int var2)
 		eventBus.post(event);
 	}
 
-	/**
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: Move the javadoc too.

File: runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java [hunk 2]
```diff
@@ -1054,4 +1055,18 @@ public static void updateNpcs(boolean var0, PacketBuffer var1)
 		// have been processed is typically more useful.
 		shouldProcessGameTick = true;
 	}
+
+	@Inject
+	@FieldHook("focused")
```
Comment (Adam-) [inline runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java]: Any reason why this isnt a methodhook like before?

File: runelite-mixins/src/main/java/net/runelite/mixins/RSItemCompositionMixin.java [hunk 3]
```diff
@@ -73,4 +76,13 @@ public void resetShiftClickActionIndex()
 	{
 		shiftClickActionIndex = DEFAULT_CUSTOM_SHIFT_CLICK_INDEX;
 	}
+
+	@Inject
+	@MethodHook(value = "post", end = true)
+	public void post()
+	{
+		final  PostItemComposition event = new PostItemComposition();
```
Comment (Adam-) [inline runelite-mixins/src/main/java/net/runelite/mixins/RSItemCompositionMixin.java]: Fix formatting

File: runelite-client/src/main/java/net/runelite/client/callback/Hooks.java [hunk 4]
```diff
@@ -383,27 +380,6 @@ public static void addChatMessage(int type, String name, String message, String
 		eventBus.post(chatMessage);
 	}
 
-	/**
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: Do not remove javadoc.

File: runelite-client/src/main/java/net/runelite/client/callback/Hooks.java [hunk 5]
```diff
@@ -367,19 +365,6 @@ public static boolean menuActionHook(int actionParam, int widgetId, int menuActi
 		return menuOptionClicked.isConsumed();
 	}
 
-	public static void addChatMessage(int type, String name, String message, String sender)
-	{
-		if (log.isDebugEnabled())
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: Any reason this is removed? The Hooks logger is accessible. We could also maybe just inject a logger.

File: runelite-mixins/src/main/java/net/runelite/mixins/RSMainBufferProviderMixin.java [hunk 6]
```diff
@@ -42,6 +42,6 @@
 	@Replace("draw")
 	final void draw(Graphics graphics, int x, int y)
 	{
-		Hooks.draw(this, graphics, x, y);
+		Hooks.draw(this, graphics);
```
Comment (Adam-) [inline runelite-mixins/src/main/java/net/runelite/mixins/RSMainBufferProviderMixin.java]: What is the point of this commit?

File: runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java (hunk 2 above)
Comment (deathbeam) [reply runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java]: No reason for it to be methodHook when that field is properly modified in both focusLost and focusGained, this felt like simpler approach.

File: runelite-mixins/src/main/java/net/runelite/mixins/RSItemCompositionMixin.java (hunk 3 above)
Comment (deathbeam) [reply runelite-mixins/src/main/java/net/runelite/mixins/RSItemCompositionMixin.java]: For the annotation?

File: runelite-mixins/src/main/java/net/runelite/mixins/RSMainBufferProviderMixin.java (hunk 6 above)
Comment (deathbeam) [reply runelite-mixins/src/main/java/net/runelite/mixins/RSMainBufferProviderMixin.java]: Consistency with other draw-related methods, it does not need x/y so only needed fields should be exposed

**Pull request merged**